tools/play_animation.py

```diff
@@ -0,0 +1,289 @@
+        return array_members
+    print("Error loading animation array. Hmm.")
+    exit()
+
+
+# Get every array of u8's and return as a Python dictionary
+def load_anims(src_file):
+    with open(src_file) as f:
+        file_lines = f.read().split("\n")
+    loaded_anims = {}
+    for i, line in enumerate(file_lines):
+        if line.startswith("static u8"):
+            # handle the potential of multi-line animations
+            anim = ""
+            line_idx = 0
+            while ";" not in anim:
+                anim += file_lines[i + line_idx]
+                line_idx += 1
+            # got full animation line. Strip spaces especially since line spacing might be weird.
+            anim = anim.replace(" ", "")
+            # Now use regex to find name and the data between the curly brackets
+            anim_name = re.findall(r"(?<=staticu8)[^\[]*", anim)[0]
+            anim_data = re.findall(r"(?<={)[^}]*", anim)[0]
+            # Turn the data into a Python list of numbers
+            anim_data = [int(x, 0) for x in anim_data.split(",")]
+            loaded_anims[anim_name] = anim_data
+    return loaded_anims
+
+
+def get_initializer_for_ent(anim_name, src_file, overlay):
+    initName = None
+    print(f"Getting animset for animation {anim_name} in {src_file}, part of {overlay}")
+    with open(src_file) as f:
+        filelines = f.readlines()
+    for i, line in enumerate(filelines):
+        if anim_name in line and "static u8" not in line:
+            # Now we've found a line where it's used. Search backward for
+            # a call to InitializeEntity.
+            print(f"Anim used in line {i}:")
+            print(line)
+            for lookback in range(i, 0, -1):
+                lbline = filelines[lookback]
+                if "InitializeEntity" in lbline:
+                    print("Best guess entity initializer:", lbline)
+                    initName = re.findall(r"(?<=\()[^\)]*", lbline)[0]
+    if initName is None:
+        # if we reach this point, we didn't find an automatic initializer
+        # Prompt the user.
+        options = []
+        for line in filelines:
+            if "InitializeEntity" in line:
+                option = re.findall(r"(?<=\()[^\)]*", line)[0]
+                if option not in options:
+                    options.append(option)
+        for i, it in enumerate(options):
+            print(f" [{i}]: {it}")
+        initializer_idx = int(input("Select argument above for InitializeEntity: "))
+        initName = options[initializer_idx]
+    # Now we know the name of the entity initializer.
+    # Fetch the value of it from the overlay e_init file.
+    with open("src/st/" + overlay + "/e_init.c") as f:
+        initlines = f.readlines()
+    for line in initlines:
+        if initName in line:
+            initSet = re.findall(r"(?<={)[^}]*", line)[0].split(",")
+            print(initSet)
+            return [int(x, 0) for x in initSet]
```

Comment:
This fails for initializers with the form `ANIMSET_OVL(0x05)`

```
  File "/home/josh/dev/sotn-decomp/tools/play_animation.py", line 117, in get_initializer_for_ent
    return [int(x, 0) for x in initSet]
            ^^^^^^^^^
ValueError: invalid literal for int() with base 0: 'ANIMSET_OVL(0x05)'
```

eg. `EInit D_us_80180AAC = {ANIMSET_OVL(0x05), 0x00, 0x24, 0x000, 0x003};`

Reply:
Yep I was playing around with sprite extraction here and got something "working" but it appeared to be wrong or not working with this tool so I'll need to dig into it further